src/map/ai/controllers/spirit_controller.cpp

```diff
@@ -0,0 +1,593 @@
+    {
+        PSpirit->m_healSpells.push_back(static_cast<uint16>(SpellID::Curaga_III));
+    }
+    if (mLvl >= 50)
+    {
+        PSpirit->m_offensiveSpells.push_back(static_cast<uint16>(SpellID::Holy));
+    }
+    if (mLvl >= 47 && mLvl < 63)
+    {
+        PSpirit->m_buffSpells.push_back(static_cast<uint16>(SpellID::Protect_III));
+    }
+    if (mLvl >= 41)
+    {
+        PSpirit->m_healSpells.push_back(static_cast<uint16>(SpellID::Cure_IV));
+    }
+    if (mLvl >= 40)
+    {
+        PSpirit->m_buffSpells.push_back(static_cast<uint16>(SpellID::Haste));
+        PSpirit->m_offensiveSpells.push_back(static_cast<uint16>(SpellID::Banishga_II));
+    }
+    if (mLvl >= 37)
+    {
+        PSpirit->m_offensiveSpells.push_back(static_cast<uint16>(SpellID::Flash));
+    }
+    if (mLvl >= 37 && mLvl < 57)
+    {
+        PSpirit->m_buffSpells.push_back(static_cast<uint16>(SpellID::Shell_II));
+    }
+    if (mLvl >= 31)
+    {
+        PSpirit->m_offensiveSpells.push_back(static_cast<uint16>(SpellID::Dia_II));
+    }
+    if (mLvl >= 31 && mLvl < 71)
+    {
+        PSpirit->m_healSpells.push_back(static_cast<uint16>(SpellID::Curaga_II));
+    }
+    if (mLvl >= 30)
+    {
+        PSpirit->m_offensiveSpells.push_back(static_cast<uint16>(SpellID::Banish_II));
+    }
+    if (mLvl >= 27 && mLvl < 57)
+    {
+        PSpirit->m_buffSpells.push_back(static_cast<uint16>(SpellID::Protect_II));
+    }
+    if (mLvl >= 21)
+    {
+        PSpirit->m_buffSpells.push_back(static_cast<uint16>(SpellID::Regen));
+    }
+    if (mLvl >= 21 && mLvl < 41)
+    {
+        PSpirit->m_healSpells.push_back(static_cast<uint16>(SpellID::Cure_III));
+    }
+    if (mLvl >= 17 && mLvl < 37)
+    {
+        PSpirit->m_buffSpells.push_back(static_cast<uint16>(SpellID::Shell));
+    }
+    if (mLvl >= 16 && mLvl < 51)
+    {
+        PSpirit->m_healSpells.push_back(static_cast<uint16>(SpellID::Curaga));
+    }
+    if (mLvl >= 11 && mLvl < 41)
+    {
+        PSpirit->m_healSpells.push_back(static_cast<uint16>(SpellID::Cure_II));
+    }
+    if (mLvl >= 7 && mLvl < 27)
+    {
+        PSpirit->m_buffSpells.push_back(static_cast<uint16>(SpellID::Protect));
+    }
+    if (mLvl >= 5 && mLvl < 65)
+    {
+        PSpirit->m_offensiveSpells.push_back(static_cast<uint16>(SpellID::Banish));
+    }
+    if (mLvl >= 1 && mLvl < 21)
+    {
+        PSpirit->m_healSpells.push_back(static_cast<uint16>(SpellID::Cure));
+    }
+}
+
+int16 CSpiritController::GetSMNSkillReduction()
+{
+    if (PSpirit->PMaster && PSpirit->PMaster->objtype == TYPE_PC)
+    {
+        uint8 masterLvl = PSpirit->PMaster->GetMLevel();
+
+        if (PSpirit->PMaster->GetMJob() != JOB_SMN)
+        {
+            masterLvl = PSpirit->PMaster->GetSLevel();
+        }
+
+        uint16 skill    = PSpirit->PMaster->GetSkill(SKILL_SUMMONING_MAGIC);
+        uint16 maxSkill = battleutils::GetMaxSkill(SKILL_SUMMONING_MAGIC, JOB_SMN, masterLvl);
+
+        return (1000 * std::clamp(skill - maxSkill, 0, 255));
```

Comment:
255?

Reply:
255 as realistically there will never be a > 255 skill difference. This was just to satisfy the clamp.